plugins/outputs/logzio/logzio.go

```diff
@@ -0,0 +1,171 @@
+	"github.com/influxdata/telegraf/plugins/outputs"
+	lg "github.com/logzio/logzio-go"
+)
+
+const (
+	defaultLogzioCheckDiskSpace = true
+	defaultLogzioDiskThreshold  = 98 // represent % of the disk
+	defaultLogzioDrainDuration  = "3s"
+	defaultLogzioURL            = "https://listener.logz.io:8071"
+
+	minDiskThreshold = 0
+	maxDiskThreshold = 100
+
+	logzioDescription = "Send aggregate metrics to Logz.io"
+	logzioType        = "telegraf"
+)
+
+var sampleConfig = `
+  ## Set to true if Logz.io sender checks the disk space before adding metrics to the disk queue.
+  # check_disk_space = true
+
+  ## The percent of used file system space at which the sender will stop queueing. 
+  ## When we will reach that percentage, the file system in which the queue is stored will drop 
+  ## all new logs until the percentage of used space drops below that threshold.
+  # disk_threshold = 98
+
+  ## How often Logz.io sender should drain the queue.
+  ## Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
+  # drain_duration = "3s"
+
+  ## Where Logz.io sender should store the queue
+  ## queue_dir = Sprintf("%s%s%s%s%d", os.TempDir(), string(os.PathSeparator),
+  ##                     "logzio-buffer", string(os.PathSeparator), time.Now().UnixNano())
+
+  ## Logz.io account token
+  token = "your logz.io token" # required
+
+  ## Use your listener URL for your Logz.io account region.
+  # url = "https://listener.logz.io:8071"
+`
+
+type Logzio struct {
+	CheckDiskSpace bool   `toml:"check_disk_space"`
+	DiskThreshold  int    `toml:"disk_threshold"`
+	DrainDuration  string `toml:"drain_duration"`
+	QueueDir       string `toml:"queue_dir"`
+	Token          string `toml:"token"`
+	URL            string `toml:"url"`
+
+	sender *lg.LogzioSender
+}
+
+type Metric struct {
+	Metric     map[string]interface{} `json:"metrics"`
+	Dimensions map[string]string      `json:"dimensions"`
+	Time       time.Time              `json:"@timestamp"`
+	Type       string                 `json:"type"`
+}
+
+func (l *Logzio) initializeSender() error {
+	if l.Token == "" || l.Token == "your logz.io token" {
+		return fmt.Errorf("[logzio] token is required")
+	}
+
+	drainDuration, err := time.ParseDuration(l.DrainDuration)
+	if err != nil {
+		return fmt.Errorf("[logzio] failed to parse drain_duration: %s", err)
+	}
+
+	diskThreshold := l.DiskThreshold
+	if diskThreshold < minDiskThreshold || diskThreshold > maxDiskThreshold {
+		return fmt.Errorf("[logzio] threshold has to be between %d and %d", minDiskThreshold, maxDiskThreshold)
+	}
+
+	l.sender, err = lg.New(
+		l.Token,
+		lg.SetCheckDiskSpace(l.CheckDiskSpace),
+		lg.SetDrainDiskThreshold(l.DiskThreshold),
+		lg.SetDrainDuration(drainDuration),
+		lg.SetTempDirectory(l.QueueDir),
+		lg.SetUrl(l.URL),
+	)
+
+	if err != nil {
+		return fmt.Errorf("[logzio] failed to create new logzio sender: %v", err)
+	}
+
+	log.Printf("I! [logzio] Successfuly created Logz.io sender: %s %s %s %d\n", l.URL, l.QueueDir,
```

Comment:
add 
```go
Log Telegraf.Logger `toml:"-"`
```
to the struct and use that instead. Telegraf will initialize it for you, and you can use testutil.Logger{} in tests.
This uses the Telegraf logger interface which will set warning level and plugin name for you.

Reply:
@ssoroka I think it wasn't part of the telegraf project when I first sent the PR. I keep getting errors when I try to use Telegraf logger locally. Any ideas why?

![image](https://user-images.githubusercontent.com/32218210/94777106-9223db80-03cb-11eb-8184-499183df51d3.png)